src/Sonarr.Api.V3/EpisodeFiles/EpisodeFileModule.cs

```diff
@@ -113,6 +115,16 @@ private object SetQuality()
                 {
                     episodeFile.Quality = resource.Quality;
                 }
+
+                if (resource.SceneName != null)
+                {
+                    episodeFile.SceneName = resource.SceneName;
```

Comment:
`SceneName` isn't a freeform field, this will need to be validated to ensure it conforms with it.

How it's calculated during import is here: https://github.com/Sonarr/Sonarr/blob/develop/src/NzbDrone.Core/MediaFiles/EpisodeImport/SceneNameCalculator.cs

There is also `OriginalFilePath` which is the job folder (if present) and the filename.

Reply:
The `isScenePath` method https://github.com/Sonarr/Sonarr/blob/d6e5d9c424b39680d169bced723b3615043b0f97/src/NzbDrone.Core/Parser/SceneChecker.cs#L7 appears to be designed to determine if a file name is also a `SceneName` during import since it may be ambiguous. Not sure if this check needs to be performed in a situation where it was explicitly being set by the API in a less ambiguous situation

Reply:
It's not the for the ambiguity, it's to determine that it is actually a scene name based on those rules. Unless @Taloth disagrees with this being validated I think it makes sense that it conforms with those checks.